app/_kong_plugins/openid-connect/index.md

```diff
@@ -37,6 +35,541 @@ search_aliases:
+within a service, developers can place {{site.base_gateway}} in front of the upstream service and have {{site.base_gateway}} handle authentication. 
+This separation allows developers to focus on the business logic within their application, easily swap out services while preserving authentication at the front door, and effortlessly spread the same authentication to new services.
+
+You may prefer OpenID Connect over other authentication types, such as Key Auth and Basic Auth, 
+because you don't need to manage a database for storing passwords. Instead, you can offload the 
+task to a trusted identity provider of your choice.
+
+While the OpenID Connect plugin suits many different use cases and extends other plugins 
+such as [JWT](/plugins/jwt/) (JSON Web Token), [ACL](/plugins/acl/), and [0Auth 2.0](/plugins/oauth2/), 
+the most common use case is the [authorization code flow](#authorization-code-flow).
+
+## Where do I begin?
+
+This plugin includes many configuration parameters that allow finely grained customization.
+Review the following important configuration parameters before implementing any flows or grants:
+
+Parameter | Purpose
+----------|--------
+`config.issuer` <br> *required* | This parameter tells the plugin where to find discovery information, and it is the only required parameter. You should set the value `realm` or `iss` on this parameter if you don't have a discovery endpoint. <br><br>This parameter doesn't have to match the URL of the `iss` claim in the access tokens being validated. To set URLs supported in the `iss` claim, use `config.issuers_allowed`.
+`config.auth_methods` | Decide what authentication grants to use with this plugin and configure the `config.auth_methods` field accordingly. To restrict the scope of potential attacks, the parameter should only contain the grants that you want to use. 
+`config.client_id` | In many cases, you need to specify `config.client_id`. Also, if your identity provider requires authentication, such as on a token endpoint, you will need to specify the client authentication credentials, for example `config.client_secret`.
+`config.audience_required` | If you are using a public identity provider, such as Google, you should limit the audience with `config.audience_required` to contain only your `config.client_id`. You may also need to adjust `config.audience_claim` in case your identity provider uses a non-standard claim (other than `aud`, as specified in JWT standard). This is important because some identity providers share public keys with different clients.
+`config.session_secret` | If you are running {{site.base_gateway}} in DB-less mode with declarative configuration and session cookie authentication, set `config.session_secret`. Leaving this parameter unset will result in every Nginx worker across your nodes encrypting and signing the cookies with their own secrets.
+
+For all available configuration parameters, see the 
```

Comment:
I see the point of the table above, but if you could kill it I would kill it in favor of better descriptions in the schema.